Dashboard/Server/SessionManagement/ServerSessionManager.cs

```diff
@@ -54,35 +106,109 @@ bool IsValidIPAddress(string IPAddress)
             }
 
             // We have 4 bytes in a address
-            byte tempForParsing;
+            //byte tempForParsing;
 
             // for each part(elements of byteValues list), we check whether the string 
             // can be successfully converted into a byte or not.
-            return byteValues.All(r => byte.TryParse(r, out tempForParsing));
+            return byteValues.All(r => byte.TryParse(r, out byte tempForParsing));
         }
 
         /// <summary>
-        /// Returns the credentials required to 
-        /// Join or start the meeting
+        /// All subscribers are notified about the new session by calling the 
+        /// OnAnalyticsChanged function
         /// </summary>
-        /// <returns> A MeetingCredentials Object </returns>
-        public MeetingCredentials GetPortsAndIPAddress()
+        public void NotifyTelemetryModule()
         {
-            ICommunicator communicator = CommunicationFactory.GetCommunicator(false);
-            Trace.WriteLine("Fetching IP Address and port from the networking module");
-            string meetAddress = communicator.Start();
+            for(int i=0;i<_telemetrySubscribers.Count;++i)
+            {
+                lock(this)
+                {
+                    _telemetrySubscribers[i].OnAnalyticsChanged(_sessionData);
+                }
+            }
+        }
 
-            if (IsValidIPAddress(meetAddress) != true)
+        /// <summary>
+        /// This function is called by the networking module when a user joins the meeting. The
+        /// SocketObject received from the networking module is then passed again but with a unique 
+        /// ID to identify the object uniquely. 
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="socketObject"></param>
+        public void OnClientJoined<T>(T socketObject)
+        {
+            lock(this)
             {
-                Trace.WriteLine("IP Address is not valid, return null");
-                return null;
+                userCount += 1;
             }
+            _communicator.AddClient<T>(userCount.ToString(), socketObject);
+        }
 
-            Trace.WriteLine("Returning the IP Address to the UX");
-            string ipAddress = meetAddress.Substring(0, meetAddress.IndexOf(':'));
-            int port = Convert.ToInt16(meetAddress.Substring(meetAddress.IndexOf(':') + 2));
+        private void ClientArrivalProcedure(ClientToServerData arrivedClient)

```

Comment:
In this function ClientArrivalProcedure, u need to call the Content's SSendAllMessagesToClient to send all the chat to the user.

Reply:
Can you explain more, I am a bit cloudy on this

Reply:
The  SSendAllMessagesToClient will send the list of chats from server to client. This is a required step in client arrival. 
The dashboard needs to call this function so that the chats are sent to the client along with the user object. The function takes the ID of the user.